src/canvas-effects/canvas-effect.js

```diff
@@ -3392,46 +3114,57 @@ export default class CanvasEffect extends PIXI.Container {
 				(creationTimeDifference % (loopDuration * 1000)) / 1000;
 		}
 
-		return this._resetLoop();
+		if (this._loopOffset) {
+			this.mediaCurrentTime = this._loopOffset
+		}
+		await this.playMedia();
+		this._addToTicker(this.loopHandler)
 	}
 
-	/**
-	 * Continuously reset the video to the right time so that the start and end time can be preserved
-	 *
-	 * @returns {Promise<void>}
-	 * @private
-	 */
-	async _resetLoop(firstLoop = true) {
-		if (this._ended) return;
-
-		let loopWaitTime = 0;
-		if (this._animationTimes.loopStart) {
-			if (this._isEnding) return;
-			this.mediaCurrentTime = (firstLoop ? 0 : this._animationTimes.loopStart) + (this._loopOffset > 0 ? this._loopOffset : 0);
-			loopWaitTime = (this._animationTimes.loopEnd - this.mediaCurrentTime) * 1000;
-		} else {
-			this.mediaCurrentTime = this._startTime + this._loopOffset;
-			loopWaitTime = this._animationDuration - this._loopOffset * 1000;
+	async loopHandler() {
```

Comment:
We may need an early exit here in case the media only actually plays once with no extra fluff. It currently causes single shot effect to play once and then play a tiny bit of the beginning before ending.

Try it with this:
```js
new Sequence()
    .effect() // Uses Sequencer database. Is not affected by template.
	.file("jb2a.melee_generic.bludgeoning.one_handed")
	.attachTo(token)
	.rotateTowards(game.user.targets.first())
    .template({
        "gridSize": 200,
        "startPoint": 150,
        "endPoint": 200
	})
    .effect() // Does not use Sequencer database. Affected by template.
	.file("modules/jb2a_patreon/Library/Generic/Weapon_Attacks/Melee/DmgBludgeoning_01_Regular_Yellow_1Handed_800x600.webm")
	.attachTo(token)
	.rotateTowards(game.user.targets.first())
    .template({
        "gridSize": 200,
        "startPoint": 150,
        "endPoint": 200
	})
    .play()
```

Reply:
This I cannot reproduce... The animation should never play from the beginning again. I even removed the end timeout that cleans up the effect after the calculated duration and in this case the animation simply stops at the last frame.

There were some issues with the removal of the tick listener once it's not needed anymore, but the animation never started again from the beginning.  What am I missing?